lib/kiev/shoryuken/store_request_details.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Kiev
+  module Shoryuken
+    class StoreRequestDetails
+      WRAPPED = "wrapped"
+
+      def call(_worker, _queue, message, parsed_body)
+        Config.instance.jobs_propagated_fields.each do |key|
+          Kiev[key] = field_value(message, parsed_body, key)
+        end
+        request_store = Kiev::RequestStore.store
+        request_store[:background_job] = true
+        request_store[:message_id] = message.message_id
+        yield
+      end
+
+      private
+
+      def field_value(message, parsed_body, key)
+        # We may want to separate the lists of keys propagated from body
+        # and message attributes; or introduce a concept of "categories",
+        # the existing ones being "http" and "jobs"
+
+        # Shoryuken's body_parser may return any container type
+        # respond_to?(:[]) may cause weird behavior on unparsed strings
+        if parsed_body.is_a?(Hash)
+          value = parsed_body[key]
```

Comment:
What is the idea behind this piece of code?

Reply:
Shoryuken has a `body_parser` parameter that can transform the message body before it's given off to middlewares. We should make use of this one by the way, it makes code slightly simpler and allows us to make use of one of Kiev's features:

Kiev has `jobs_propagated_keys` configuration entry to include additional fields from the job into the logs, However, there's a little problem with this: Sidekiq only has one JSON payload, SQS has two: message attributes and the message's body itself.

The initial thought was to include the `"event"` key using only Kiev's configuration (that'd be neat), but then I discovered that that key is already used by Kiev itself, and it doesn't support aliasing/renaming. So we won't make use of this feature anyway, resorting to assigning fields inside consumers' code.

At this point we can dump support for additional fields from body entirely. We won't use it, other users can use `Kiev[key] = value` in consumers.